fdbserver/workloads/AuthzSecurity.actor.cpp

```diff
@@ -0,0 +1,409 @@
+		tLogConfigKey = getOption(options, "tLogConfigKey"_sr, "TLogInterface"_sr);
+		ASSERT(g_network->isSimulated());
+		// make it comfortably longer than the timeout of the workload
+		signedToken = g_simulator->makeToken(
+		    tenant, uint64_t(std::lround(getCheckTimeout())) + uint64_t(std::lround(testDuration)) + 100);
+		signedTokenAnotherTenant = g_simulator->makeToken(
+		    anotherTenant, uint64_t(std::lround(getCheckTimeout())) + uint64_t(std::lround(testDuration)) + 100);
+		testFunctions.push_back(
+		    [this](Database cx) { return testCrossTenantGetDisallowed(PositiveTestcase::True, cx, this); });
+		testFunctions.push_back(
+		    [this](Database cx) { return testCrossTenantGetDisallowed(PositiveTestcase::False, cx, this); });
+		testFunctions.push_back(
+		    [this](Database cx) { return testCrossTenantCommitDisallowed(PositiveTestcase::True, cx, this); });
+		testFunctions.push_back(
+		    [this](Database cx) { return testCrossTenantCommitDisallowed(PositiveTestcase::False, cx, this); });
+		testFunctions.push_back(
+		    [this](Database cx) { return testPublicNonTenantRequestsAllowedWithoutTokens(cx, this); });
+		testFunctions.push_back([this](Database cx) { return testTLogReadDisallowed(cx, this); });
+	}
+
+	Future<Void> setup(Database const& cx) override { return Void(); }
+	Future<Void> start(Database const& cx) override {
+		for (int c = 0; c < actorCount; c++)
+			clients.push_back(timeout(runTestClient(cx, this), testDuration, Void()));
+		return waitForAll(clients);
+	}
+	Future<bool> check(Database const& cx) override {
+		int errors = 0;
+		for (int c = 0; c < clients.size(); c++)
+			errors += clients[c].isError();
+		if (errors)
+			TraceEvent(SevError, "TestFailure").detail("Reason", "There were client errors.");
+		clients.clear();
+		return errors == 0 && crossTenantGetPositive.getValue() > 0 && crossTenantGetNegative.getValue() > 0 &&
+		       crossTenantCommitPositive.getValue() > 0 && crossTenantCommitNegative.getValue() > 0 &&
+		       publicNonTenantRequestPositive.getValue() > 0;
+	}
+	void getMetrics(std::vector<PerfMetric>& m) override {
+		m.push_back(crossTenantGetPositive.getMetric());
+		m.push_back(crossTenantGetNegative.getMetric());
+		m.push_back(crossTenantCommitPositive.getMetric());
+		m.push_back(crossTenantCommitNegative.getMetric());
+		m.push_back(publicNonTenantRequestPositive.getMetric());
+		m.push_back(tLogReadNegative.getMetric());
+	}
+
+	void setAuthToken(Transaction& tr, Standalone<StringRef> token) {
+		tr.setOption(FDBTransactionOptions::AUTHORIZATION_TOKEN, token);
+	}
+
+	ACTOR Future<Void> setKeyValue(Transaction* tr, StringRef key, StringRef value) {
+		loop {
+			try {
+				tr->set(key, value);
+				wait(tr->commit());
+				break;
+			} catch (Error& e) {
+				wait(tr->onError(e));
+			}
+		}
+		return Void();
+	}
+
+	ACTOR Future<KeyRangeLocationInfo> extractCachedLocation(Transaction* tr,
```

Comment:
This actor doesn't do what its name suggests. It gets the key location whether it is cached or not.

I think this should be renamed and take a Database and not a transaction as argument

Reply:
Will do.